release/_bom.sh

```diff
@@ -224,58 +224,45 @@ function generate_pom_release_dxp_bom_third_party {
 		-e "w ${pom_file_name}" \
 		"${_RELEASE_TOOL_DIR}/templates/release.dxp.bom.third.party.pom.tpl" > /dev/null
 
-	local property_files=(
-		"${_PROJECTS_DIR}/liferay-portal-ee/lib/development/dependencies.properties"
-		"${_PROJECTS_DIR}/liferay-portal-ee/lib/portal/dependencies.properties"
-	)
+	IFS=$'\n' read -d '' -ra dependencies_properties < "${_PROJECTS_DIR}/liferay-portal-ee/lib/development/dependencies.properties"
+	IFS=$'\n' read -d '' -ra portal_dependencies_properties < "${_PROJECTS_DIR}/liferay-portal-ee/lib/portal/dependencies.properties"
 
-	find "${_BUNDLES_DIR}/tomcat/webapps/ROOT/WEB-INF/shielded-container-lib" -name "*.jar" -print0 | \
-		while IFS= read -d '' -r jar_file
-		do
-			if (unzip -l "${jar_file}" | grep -q "pom.xml$")
-			then
-				local artifact_name="${jar_file##*/}"
-
-				artifact_name="${artifact_name/%\.jar}"
-
-				if [[ "${artifact_name}" == com.liferay.* ]]
-				then
-					continue
-				fi
-
-				artifact_name="${artifact_name##*.}"
-
-				local artifact_properties=$(grep -w "^$artifact_name" "${property_files[@]}")
-
-				if [ -z "${artifact_properties}" ]
-				then
-					continue
-				fi
-
-				if [[ "${artifact_properties}" == *development/dependencies.properties* ]] && [[ "${artifact_properties}" == *portal/dependencies.properties* ]]
-				then
-					artifact_properties=$(echo "${artifact_properties}" | grep "portal/dependencies.properties" | cut -d= -f2)
-				else
-					artifact_properties=$(echo "$artifact_properties" | cut -d= -f2)
-				fi
-
-				echo "$artifact_properties" | \
-					while IFS=: read -r group_id artifact_id version
-					do
-						echo -e "\t\t\t<dependency>"
-						echo -e "\t\t\t\t<groupId>${group_id}</groupId>"
-						echo -e "\t\t\t\t<artifactId>${artifact_id}</artifactId>"
-						echo -e "\t\t\t\t<version>${version}</version>"
-						echo -e "\t\t\t</dependency>"
-					done >> "${pom_file_name}"
-			fi
-		done
+	dependencies_properties+=("${portal_dependencies_properties[@]}")
+
+	local pom_compile_only_file_name="release.dxp.bom.compile.only-${_PRODUCT_VERSION}-${_BUILD_TIMESTAMP}.pom"
+
+	local dependencies_included=()
+
+	for dependency_property in "${dependencies_properties[@]}"
+	do
+		IFS=':' read -ra dependency_property_parts <<< "$(echo "${dependency_property}" | cut -d = -f 2)"
+
+		if [[ ${dependencies_included[@]} =~ "${dependency_property_parts[0]}${dependency_property_parts[1]}${dependency_property_parts[2]}" ]]
```

Comment:
@lucasmiranda0, I just realized now, why don't you include here the `dependency_property` instead of `${dependency_property_parts[0]}${dependency_property_parts[1]}${dependency_property_parts[2]}`?

I believe it'll improve the code legibility, what do you think?

Reply:
Yeah, I just updated.